types/react-dnd/index.d.ts

```diff
@@ -0,0 +1,37 @@
+declare module ReactDnD {
+    type ReactProps = any;
+
+    type _DragSourceTypePrimitives = string | symbol;
+    type DragSourceType = _DragSourceTypePrimitives | ((props: ReactProps) => _DragSourceTypePrimitives);
+
+    type _DropTargetTypesPrimitives = string | symbol | Array<string | symbol>;
+    type DropTargetTypes = _DropTargetTypesPrimitives | ((props: ReactProps) => _DropTargetTypesPrimitives);
+
+    interface IDragSource {
+        beginDrag: (props: ReactProps, monitor: DragSourceMonitor, component: YourReactClass) => any;

```

Comment:
Where does `YourReactClass` come from?

Reply:
Good question, I took it from their website. I replaced with `React.Component`